### PR TITLE
Lazy-load some js libraries

### PR DESCRIPTION
**Pull request by tiberiuichim**
WIP on lazy-loading external libraries.

### CODE REVIEW

**Comment by tiberiuichim** [comment]
I'll give up on the current PR. The problems are:

- Although it's somewhat nice to be able to say `withLoadables(['toastify', 'Select'])` and such, I feel that there's a level of indirection, specially as there should be some level of inteligence in the component that uses the lazy component. I don't think we can provide a solution for all of the components, how they should behave when the lazy component hasn't been loaded yet.
- All tests then need to include a mock. We can automock it, but they still need to `beforeAll` force the lazy components to load.

It's been a great learning opportunity, but I'll pass for now. I'll try to use this knowledge to further enhance the HTML block, that one is a bit of a pain to lazy-load because of the Prism library.

**Comment by nileshgulia1** [comment]
@tiberiuichim I see the following traceback when creating a page. 
Screenshot 2020-11-30 at 2 50 00 PM

File: src/components/manage/Widgets/SelectWidget.jsx [hunk 1]
```diff
@@ -242,6 +243,9 @@ class SelectWidget extends Component {
    */
   render() {
     const { id, choices, value, onChange } = this.props;
+    // const { Select, AsyncPaginate } = this.props;
```
Comment (tiberiuichim) [inline src/components/manage/Widgets/SelectWidget.jsx:246]: Cleanup

File: src/components/manage/Widgets/SelectWidget.jsx [hunk 2]
```diff
@@ -318,6 +322,7 @@ class SelectWidget extends Component {
 
 export default compose(
   injectIntl,
+  // withLoadables(['Select', 'AsyncPaginate']),
```
Comment (tiberiuichim) [inline src/components/manage/Widgets/SelectWidget.jsx:325]: Cleanup

File: src/components/manage/Widgets/SelectWidget.test.jsx [hunk 3]
```diff
@@ -22,17 +28,21 @@ test('renders a select widget component', async () => {
...
+        <SelectWidget
+          id="my-field"
+          title="My field"
+          onChange={() => {}}
+          onBlur={() => {}}
+          onClick={() => {}}
+        />
+      </Provider>,
+    );
+  });
```
Comment (tiberiuichim) [inline src/components/manage/Widgets/SelectWidget.test.jsx:45]: TODO: use the more explicit style to test, see Workflow.test.jsx

File: src/reducers/lazyLibraries/lazyLibraries.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { LOAD_LAZY_LIBRARY } from '@plone/volto/constants/ActionTypes';
+
+const initialState = {};
+
+export default function lazyLibraries(state = initialState, action = {}) {
```
Comment (tiberiuichim) [inline src/reducers/lazyLibraries/lazyLibraries.js:5]: Add docstring

**Comment by tiberiuichim** [comment]
TODO on this PR:

- fix the HTML block test (the snapshot was wrongly updated)
- provide "bundles", a way to preload a set of libraries. Instead of loading the extra libs "on demand" and be faced with empty components, why not preload them when we login, or when faced with the Edit screen?

**Comment by tiberiuichim** [comment]
TODO:

- add ref forwarding
- rename the "withLoadables" to "injectLazyLib"
- provide a "preload" hoc/hook pair that can preload some libraries

**Comment by tiberiuichim** [comment]
Rename to injectLazyLibs

**Pull request merged**